lib/utils.js

```diff
@@ -78,7 +78,7 @@ function filterHeaders (headers, filter) {
 
 function buildURL (source = '', reqBase) {
   // issue ref: https://github.com/fastify/fast-proxy/issues/42
-  const cleanSource = source.replace(/\/+/g, '/')
+  const cleanSource = source.replace(/^\/+/g, '/')
```

Comment:
The regexp is no more necessary - it can be converted to be more performant:

https://www.nearform.com/blog/the-secrets-behind-high-performance-with-node-js/

```suggestion
  let i = 0
  while (source[i] === '/') {
    i++
  }
  const cleanSource = '/' + source.substring(i)
```

Reply:
Clever, I like it. I'll leave it to the maintainers to decide if they'd rather have this or a regex.

Reply:
Oh, you are a maintainer. I'll make the change!